circuit-benchmarks/src/tx_circuit.rs

```diff
@@ -0,0 +1,109 @@
+//! Tx circuit benchmarks
+
+#[cfg(test)]
+mod tests {
+    use crate::bench_params::DEGREE;
+    use ark_std::{end_timer, start_timer};
+    use env_logger::Env;
+    use group::{Curve, Group};
+    use halo2_proofs::arithmetic::{BaseExt, CurveAffine, Field};
+    use halo2_proofs::plonk::{create_proof, keygen_pk, keygen_vk, verify_proof, SingleVerifier};
+    use halo2_proofs::{
+        pairing::bn256::{Bn256, Fr, G1Affine},
+        poly::commitment::{Params, ParamsVerifier},
+        transcript::{Blake2bRead, Blake2bWrite, Challenge255},
+    };
+    use rand::SeedableRng;
+    use rand_xorshift::XorShiftRng;
+    use secp256k1::Secp256k1Affine;
+    use std::marker::PhantomData;
+    use zkevm_circuits::tx_circuit::{
+        sign_verify::{SignVerifyChip, POW_RAND_SIZE, VERIF_HEIGHT},
+        TxCircuit,
+    };
+
+    #[cfg_attr(not(feature = "benches"), ignore)]
+    #[test]
+    fn bench_tx_circuit_prover() {
+        env_logger::Builder::from_env(Env::default().default_filter_or("debug")).init();
+
+        // Approximate value, adjust with changes on the TxCircuit.
+        const ROWS_PER_TX: usize = 175_000;
+        const MAX_TXS: usize = 2_usize.pow(DEGREE as u32) / ROWS_PER_TX;
+        const MAX_CALLDATA: usize = 1024;
+
+        let mut rng = XorShiftRng::from_seed([
+            0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06,
+            0xbc, 0xe5,
+        ]);
+        let aux_generator =
+            <Secp256k1Affine as CurveAffine>::CurveExt::random(&mut rng).to_affine();
+        let chain_id: u64 = 1337;
+        let txs = Vec::new();
+
+        let randomness = Fr::random(&mut rng);
+        let mut instances: Vec<Vec<Fr>> = (1..POW_RAND_SIZE + 1)
+            .map(|exp| vec![randomness.pow(&[exp as u64, 0, 0, 0]); txs.len() * VERIF_HEIGHT])
+            .collect();
+        // SignVerifyChip -> ECDSAChip -> MainGate instance column
+        instances.push(vec![]);
+        let circuit = TxCircuit::<Fr, MAX_TXS, MAX_CALLDATA> {
+            sign_verify: SignVerifyChip {
+                aux_generator,
+                window_size: 2,
+                _marker: PhantomData,
+            },
+            randomness,
+            txs,
+            chain_id,
+        };
+
+        // Bench setup generation
+        let setup_message = format!(
+            "Setup generation with degree = {} (MAX_TXS = {})",
+            DEGREE, MAX_TXS
+        );
+        let start1 = start_timer!(|| setup_message);
+        let general_params: Params<G1Affine> =
+            Params::<G1Affine>::unsafe_setup::<Bn256>(DEGREE.try_into().unwrap());
+        let verifier_params: ParamsVerifier<Bn256> = general_params.verifier(DEGREE * 2).unwrap();
```

Comment:
Thanks for explaining this! I've updated the test so that it uses 1 tx.  I've also updated the instance columns to depend on `MAX_TXS` and not `txs.len()`, because the public input length should be fixed per circuit, whereas `txs.len()` is variable (the circuit has a `MAX_TXS` capacity and accepts "padding" txs)

Eventhough the last bench github action failed, the bench passed with these results:
```
Start:   Setup generation with degree = 19 (MAX_TXS = 2)
End:     Setup generation with degree = 19 (MAX_TXS = 2) ...........................4.259s
test tx_circuit::tests::bench_tx_circuit_prover has been running for over 60 seconds
Start:   Tx Proof generation with 19 degree
End:     Tx Proof generation with 19 degree ........................................290.107s
Start:   Tx Proof verification
End:     Tx Proof verification .....................................................23.532ms
```
The reason why the github action failed is because the current bench scripts don't expect a previous bench result, but in this case there was one from previous runs.